### PR TITLE
Add teacher1920 application email support

### PR DESCRIPTION
**Pull request by aoby**
- enable cc
 - add new email types and update copy on old ones ([spec](https://docs.google.com/document/d/1aDSflSE-O0A7IulHxxV7kmVAGJRH8qmqZvPXXPN5WH0/edit#))
 - add email queue to track all application emails we queue up and when we send them
 - add status log to track all application status updates and their times
 - update cronjob to use new email queue and run every morning at 9am Pacific ([16:00 UTC](https://crontab.guru/#0_16_*_*_*))
 - remove all old `send_decision_notification_email` code

Remaining work: decide whether we want to keep the old `pd_applications.decision_notification_email_sent_at` column now that it's not used. Do we want to migrate those dates into the new email queue as sent? My vote is to just delete it. We already only tracked the most recent email, and I can't imagine we'd need to audit that from last year's applications. Even if we do, we still have the full email record (though harder to filter) in `pegasus poste_deliveries`.

### CODE REVIEW

**Comment by aoby** [comment]
[Automated application emails (8)](https://trello.com/c/16TgYYdD/726-automated-application-emails-8)

**Comment by aoby** [comment]
[Cron job that sends application emails should send every day at 9am PDT (instead of every Monday)](https://trello.com/c/S61AHVOK/725-cron-job-that-sends-application-emails-should-send-every-day-at-9am-pdt-instead-of-every-monday)

**Comment by aoby** [comment]
[Cron job that sends application emails should send every day at 9am PDT (instead of every Monday)](https://trello.com/c/S61AHVOK/725-cron-job-that-sends-application-emails-should-send-every-day-at-9am-pdt-instead-of-every-monday)

**Comment by aoby** [comment]
[Support cc when sending automated emails](https://trello.com/c/BSxnB4E9/727-support-cc-when-sending-automated-emails)

**Comment by aoby** [comment]
[Support cc when sending automated emails](https://trello.com/c/BSxnB4E9/727-support-cc-when-sending-automated-emails)

File: dashboard/app/controllers/api/v1/pd/applications_controller.rb [hunk 1]
```diff
@@ -80,11 +80,20 @@ def quick_view
       end
     end
 
+    COHORT_VIEW_STATUSES = %w(
```
Comment (clareconstantine) [inline dashboard/app/controllers/api/v1/pd/applications_controller.rb:83]: should this include `paid`?

File: dashboard/app/models/pd/application/application_base.rb [hunk 2]
```diff
@@ -112,21 +112,75 @@ def set_type_and_year
...
+      self.accepted_at = accepted? ? Time.now : nil
+    end
+
+    # Queues an email for this application
+    # @param email_type [String] specifies the mailer action
+    # @param deliver_now [Boolean] (default false)
+    #   When true, send the email immediately.
+    #   Otherwise, it will remain unsent in the queue until the next morning's cronjob.
+    # @see Pd::Application::Email
+    def queue_email(email_type, deliver_now: false)
```
Comment (clareconstantine) [inline dashboard/app/models/pd/application/application_base.rb:138]: 🙌 super helpful comment

File: dashboard/app/views/pd/application/teacher1920_application_mailer/declined.html.haml [hunk 3]
```diff
@@ -3,41 +3,18 @@
...
-    to accept all applicants.
-- else
-  %p
-    Our current program model prioritizes candidates in areas where Code.org has
-    established Regional Partners who provide continued learning support
-    throughout the year. Unfortunately, without a local Partner to facilitate your
-    professional development, we cannot offer you a space in the full Professional
-    Learning Program experience this year.
+  Our team appreciated the opportunity to review your application and learn about your
+  commitment to computer science education. I'm writing to let you know we are currently
```
Comment (clareconstantine) [inline dashboard/app/views/pd/application/teacher1920_application_mailer/declined.html.haml]: tiny: I think this sentence sounds better with the word 'that': "I'm writing to let you know *that* we are currently..."

File: dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# This can be viewed on non-production environments at /rails/mailers/pd/teacher_application_mailer
+class Pd::Teacher1920ApplicationMailerPreview < ActionMailer::Preview
+  include FactoryGirl::Syntax::Methods
+
+  def confirmation__with_partner
```
Comment (clareconstantine) [inline dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb]: is the double underscore intentional?

File: dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    Pd::Application::Teacher1920ApplicationMailer.confirmation build_application
+  end
+
+  def confirmation__without_partner
```
Comment (clareconstantine) [inline dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb]: another double underscore

File: dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb (hunk 4 above)
Comment (aoby) [reply dashboard/test/mailers/previews/pd_teacher1920_application_mailer_preview.rb]: Yes, it's a convention I started using back in the workshop mail preview (see https://github.com/code-dot-org/code-dot-org/blob/26d211d9627aca201d25ec55d7ee4c767fb0a261/dashboard/test/mailers/previews/pd_workshop_mailer_preview.rb#L17), to clearly differentiate the mailer action from modifiers on the context.

The name breaks down like this: `mailer_action_name(__data_modifier)`
So in this case, the mailer action is [confirmation](https://github.com/code-dot-org/code-dot-org/pull/24950/files#diff-75a50d6eac29f50edcd70d5cc7a5e5bcR5), and it has 2 previews available: one `with_partner` and one `without_partner`. Those each describe a different context for the same mailer action.

**Pull request merged**